backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/MakeUniqueLocationNameCommand.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2021 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package org.compiere.model;
+
+import com.google.common.collect.ImmutableList;
+import de.metas.location.CountryId;
+import de.metas.location.ICountryDAO;
+import de.metas.util.Check;
+import de.metas.util.Services;
+import de.metas.util.StringUtils;
+import lombok.Builder;
+import lombok.NonNull;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public final class MakeUniqueLocationNameCommand
+{
+	private final ICountryDAO countriesRepo = Services.get(ICountryDAO.class);
+
+	private final String nameInitial;
+	private final String companyName;
+	private final I_C_Location address;
+	private final List<String> existingNames;
+	private final int maxLength;
+
+	@Builder
+	private MakeUniqueLocationNameCommand(
+			@Nullable final String name,
+			@Nullable final String companyName,
+			@NonNull final I_C_Location address,
+			@Nullable final List<String> existingNames,
+			final int maxLength)
+	{
+		this.companyName = companyName;
+		this.address = address;
+		this.existingNames = existingNames != null ? existingNames : ImmutableList.of();
+		this.maxLength = maxLength > 0 ? maxLength : Integer.MAX_VALUE;
+
+		if (Check.isEmpty(name, true) || ".".equals(name))
+		{
+			this.nameInitial = null;
+		}
+		else
+		{
+			this.nameInitial = name.trim();
+		}
+	}
+
+	public String execute()
+	{
+		final String name = !Check.isEmpty(this.nameInitial, true)
```

Comment:
i guess nameInitial will never be null here

Reply:
I took this exactly as it was in the ``MProductLocation``. I thought I it's cleaner to move it out of there  if I need it to be public.

![image](https://user-images.githubusercontent.com/16001442/146421289-4075d062-d26f-4106-9723-7d405615ebaa.png)

Reply:
agreed to leave it like it is for now